File: 1085.py
val = map(int,raw_input().split(' '))
x=val[0]
y=val[1]
w=val[2]
h=val[3]
vmin = 0
hmin = 0
if((w/2)<x):
	vmin = w-x
else:
	vmin = x
if((h/2)<y):
	hmin = h-y
else:
	hmin = y
print min(vmin,hmin)

File: 1085.cc
#include <iostream>
#include <algorithm>
using namespace std;
int main(){
	int x,y,w,h;
	int vmin,hmin;
	cin >>x>>y>>w>>h;

	if((w/2) < x){
		vmin = w-x;
	}
	else{
		vmin = x;
	}
	if((h/2) < y){
		hmin = h-y;
	}
	else{
		hmin = y;
	}
	cout << min(vmin,hmin) << endl;
	return 0;
}
